packages/node/src/serverless-functions/serverless-handler.ts

```diff
@@ -1,58 +1,74 @@
+import { addHelpers } from './helpers';
+import { create as createTsCompiler } from 'ts-node';
+import { createServer, ServerResponse } from 'http';
+// @ts-expect-error
+import { dynamicImport } from './dynamic-import.js';
 import { IncomingMessage } from 'http';
-import { Readable } from 'stream';
-import type { Bridge } from '@vercel/node-bridge/bridge';
-import { getVercelLauncher } from '@vercel/node-bridge/launcher.js';
-import { VercelProxyResponse } from '@vercel/node-bridge/types';
-
-function rawBody(readable: Readable): Promise<Buffer> {
-  return new Promise((resolve, reject) => {
-    let bytes = 0;
-    const chunks: Buffer[] = [];
-    readable.on('error', reject);
-    readable.on('data', chunk => {
-      chunks.push(chunk);
-      bytes += chunk.length;
-    });
-    readable.on('end', () => {
-      resolve(Buffer.concat(chunks, bytes));
-    });
+import { serializeRequest } from '../utils';
+import exitHook from 'exit-hook';
+import fs from 'fs';
+import listen from 'async-listen';
+import path from 'path';
+import undici from 'undici';
+
+type ServerlessServerOptions = {
+  shouldAddHelpers: boolean;
+  useRequire: boolean;
+};
+
+type ServerlessFunctionSignature = (
+  req: IncomingMessage | VercelRequest,
+  res: ServerResponse | VercelResponse
+) => void;
+
+async function createServerlessServer(
+  userCode: ServerlessFunctionSignature,
+  options: ServerlessServerOptions
+) {
+  const server = createServer((req, res) => {
+    if (options.shouldAddHelpers) addHelpers(req, res);
+    return userCode(req, res);
   });
+  exitHook(server.close);
+  return { url: await listen(server) };
 }
 
 export async function createServerlessEventHandler(
-  entrypoint: string,
-  options: {
-    shouldAddHelpers: boolean;
-    useRequire: boolean;
-  }
+  entrypointPath: string,
+  options: ServerlessServerOptions
 ): Promise<(request: IncomingMessage) => Promise<VercelProxyResponse>> {
-  const launcher = getVercelLauncher({
-    entrypointPath: entrypoint,
-    helpersPath: './helpers.js',
-    shouldAddHelpers: options.shouldAddHelpers,
-    useRequire: options.useRequire,
-
-    // not used
-    bridgePath: '',
-    sourcemapSupportPath: '',
-  });
-  const bridge: Bridge = launcher();
+  const extension = path.extname(entrypointPath);
 
-  return async function (request: IncomingMessage) {
-    const body = await rawBody(request);
-    const event = {
-      Action: 'Invoke',
-      body: JSON.stringify({
-        method: request.method,
-        path: request.url,
-        headers: request.headers,
-        encoding: 'base64',
-        body: body.toString('base64'),
-      }),
-    };
+  let userCode;
+  if (extension.startsWith('.ts')) {
+    const { compile } = createTsCompiler();
+    const content = fs.readFileSync(entrypointPath, 'utf8');
+    const filename = path.basename(entrypointPath);
+    userCode = eval(compile(content, filename)) as ServerlessFunctionSignature;
```

Comment:
`eval` always worries me. Thought we had eslint rules prevent it too. Can you add a comment here explaining what is being passed to `eval`, and maybe include why it _isnt_ a security concern?

Reply:
The `eval` is necessary there for turning the user code in a real javascript function